software/speller/data_collection_platform/dcp/api.py

```diff
@@ -38,38 +39,128 @@ def index():
+        p = Process(target=stream_bci_api, args=(subprocess_dict,))
+        # need to start process before referencing it to obtain the right process_id
         p.start()
-        shared.bci_processes_states[p.pid] = True
+        shared.bci_processes_states[p.pid] = subprocess_dict
+    while subprocess_dict['state'] != 'ready' or subprocess_dict['bci_config'] == None:
+        print('BCI NOT READY YET')
+        time.sleep(1)
+    config = OpenBCIConfig(configuration=subprocess_dict['bci_config'])
+    db.session.add(config)
+    db.session.commit()
+    subprocess_dict['config_id'] = config.id
+    # once the subprocess is ready, return from call
     return {"data": {"pid": p.pid}}, 201
 
 
-@bp.route('/openbci/<int:process_id>/ready', methods=['GET'])
-def openbci_ready(process_id: int):
-    # NOTE: `pid` is currently ignored, see TODO in openbci_stop.
-    with shared.is_bci_ready.get_lock():
-        return {"data": bool(shared.is_bci_ready.value)}, 200
-
+@bp.route('/openbci/<int:process_id>/collect/start', methods=['POST'])
+def openbci_process_collect_start(process_id:int):
+    data = request.form
+    expected_keys = ['character', 'phase' ,'frequency']
+    for key in expected_keys:
+        if key not in request.form.keys():
+            return {'Invalid Request': 'Did not contain all attributes, make sure you send character, phase and frequency'}, 400
+    # We now know that the request contains all the keys
+    if process_id not in shared.bci_processes_states:
+        return {'Invalid Request': 'There is no process with this id, make sure your process id is valid'}, 404
+    subprocess_dict = shared.bci_processes_states[process_id]
+    try:
+        subprocess_dict['character'] = data.get('character')
+        subprocess_dict['frequency'] = float(data.get('frequency'))
+        subprocess_dict['phase'] = float(data.get('phase'))
+    except:
+        return {'Invalid Request': 'Could not convert post form data. Make sure the data is the correct type. (string for character, and float for phase and frequency)'}, 400
+    
+    subprocess_dict['state'] = 'collect'
+    return {'Success': 'BCI is collecting'}, 201
+
+@bp.route('/openbci/<int:process_id>/collect/stop', methods=['POST'])
+def openbci_process_collect_stop(process_id:int):
+    if process_id not in shared.bci_processes_states:
+        return {'Invalid Request': 'There is no process with this id, make sure your process id is valid'}, 404
+    subprocess_dict = shared.bci_processes_states[process_id]
+    subprocess_dict['state'] = 'ready'
+
+    # write to database 
+    if not write_stream_data(subprocess_dict):
+        return {'result': "Did not write any new information to database, stopping collection"}, 202
+    # clear the subprocess_dict
+    subprocess_dict['character'] = None
+    subprocess_dict['frequency'] = None
+    subprocess_dict['phase'] = None
+    subprocess_dict['q'] = deque()
```

Comment:
do we need to set these values to default ? Should we simply delete subprocess_dict, e.g. `del subprocess_dict` and `del shared.bci_processes_states[p.pid]`